### PR TITLE
[devel/refactor] fix attach crash

### LINKED ISSUE #2911: [devel/refactor] Use after free
Happens when I change mailboxes via sidebar

terminal output gets a bit garbled, this is what I can recover
```
...
4 in mx_mbox_open /usr/local/src/github.com/neomutt/neomutt/mx.c:387

#7 0x560a61b1016e in change_folder_mailbox index/index.c:650
#8 0x560a61b30112 in op_sidebar_open index/functions.c:2790
#9 0x560a61b308a4 in index_function_dispatcher index/functions.c:2892
#10 0x560a61b15aa5 in mutt_index_menu index/index.c:1339
#11 0x560a61aa17d4 in main /usr/local/src/github.com/neomutt/neomutt/main.c:1255
#12 0x7f2632404b24 in __libc_start_main (/usr/lib/libc.so.6+0x27b24)

SUMMARY: AddressSanitizer: heap-use-after-free index/ibar.c:247 in ibar_index_observer
```

### PR DESCRIPTION
**Pull request by flatcap**
May fix #2911 and #2912

### CODE REVIEW

**Comment by ngortheone** [comment]
@flatcap it still crashes with use-after free, but in a different place now.

Now I can go back and forth between pager <-> attachemnt menu <-> pager
But as soon as I want to close pager and go back to index it crashes

```
heap-use-after-free index/ibar.c:292 in ibar_data_free
```

**Comment by ngortheone** [comment]
@flatcap 
WRT #2911 

Changing folders with sidebar now gives different line

```
   SUMMARY: AddressSanitizer: heap-use-after-free index/ibar.c:247 in ibar_index_observer
```

**Comment by flatcap** [comment]
Thanks, I'll have a deeper look, with ASAN, tomorrow.

File: mutt_attach.c [hunk 1]
```diff
@@ -671,8 +671,8 @@ int mutt_view_attachment(FILE *fp, struct Body *a, enum ViewAttachMode mode,
     pview.mode = PAGER_MODE_ATTACH;
     if (win->type == WT_INDEX)
     {
-      struct IndexSharedData *index_shared = dialog_find(win)->wdata;
-      index_shared->email = e;
+      // struct IndexSharedData *index_shared = dialog_find(win)->wdata;
```
Comment (ngortheone) [inline mutt_attach.c:674]: just delete these 2 lines, it was me doing something I didn't understand

Comment (flatcap) [reply mutt_attach.c:674]: will do

**Pull request merged**